src/include/sof/audio/dcblock/dcblock.h

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Google LLC. All rights reserved.
+ *
+ * Author: Sebastiano Carlucci <scarlucci@google.com>
+ */
+
+#ifndef __SOF_AUDIO_DCBLOCK_DCBLOCK_H__
+#define __SOF_AUDIO_DCBLOCK_DCBLOCK_H__
+
+#include <stdint.h>
+#include <sof/platform.h>
+#include <ipc/stream.h>
+
+struct audio_stream;
+struct comp_dev;
+
+struct dcblock_state {
+	int32_t x_prev; /**< state variable referring to x[n-1] */
+	int32_t y_prev; /**< state variable referring to y[n-1] */
+};
+
+/**
+ * \brief Type definition for the processing function for the
+ * DC Blocking Filter.
+ */
+typedef void (*dcblock_func)(const struct comp_dev *dev,
+			     const struct audio_stream *source,
+			     const struct audio_stream *sink,
+			     uint32_t frames);
+
+/* DC Blocking Filter component private data */
+struct comp_data {
+	/**< filters state */
+	struct dcblock_state state[PLATFORM_MAX_CHANNELS];
+
+	/** coefficients for the processing function */
+	int32_t R_coeffs[PLATFORM_MAX_CHANNELS];
+
+	enum sof_ipc_frame source_format;
+	enum sof_ipc_frame sink_format;
+	dcblock_func dcblock_func; /**< processing function */
+};
+
+/** \brief DC Blocking Filter processing functions map item. */
+struct dcblock_func_map {
+	enum sof_ipc_frame src_fmt; /**< source frame format */
+	dcblock_func func; /**< processing function */
+};
+
+/** \brief Map of formats with dedicated processing functions. */
+extern const struct dcblock_func_map dcblock_fnmap[];
+
+/** \brief Number of processing functions. */
+extern const size_t dcblock_fncount;
+
+/**
+ * \brief Retrieives DC Blocking processing function.
+ * \param[in,out] dev DC Blocking Filter base component device.
+ */
+static inline dcblock_func dcblock_find_func(enum sof_ipc_frame src_fmt)
```

Comment:
Doxygen error, @sebcarlucci  can you please fix?

```
sof/src/include/sof/audio/dcblock/dcblock.h:58: warning: argument 'dev' of command @param is not found in the argument list of dcblock_find_func(enum sof_ipc_frame src_fmt)
sof/src/include/sof/audio/dcblock/dcblock.h:61: warning: The following parameters of dcblock_find_func(enum sof_ipc_frame src_fmt) are not documented:
  parameter 'src_fmt'
```

To run doxygen:
```
cd sof/
cmake -S doc -B doxy
make -C doxy doc VERBOSE=1
```
Longer story:
https://thesofproject.github.io/latest/contribute/process/docbuild.html

If/when PR #2741 is merged doxygen issues will be reported by CI automatically.

Reply:
PR #2741 was merged faster than I expected. So this doxygen error is now the only one turning Travis to red on every PR.
@sebcarlucci if you're busy I can fix this myself, however I just need you (or anyone really) to tell me what to write for `src_fmt`.

Reply:
I can work on this today

Reply:
Already fixed by @sebcarlucci  in PR #2746, thx!